src/science/canopy/cbl_dryLeaf.F90

```diff
@@ -509,27 +518,19 @@ SUBROUTINE dryLeaf( dels, rad, rough, air, met,                                &
 
              ELSE
 
-                IF (ecx(i) > 0.0 .AND. canopy%fwet(i) < 1.0) THEN
-                   evapfb(i) = ( 1.0 - canopy%fwet(i)) * REAL( ecx(i) ) *dels      &
-                        / air%rlam(i)
-
-                   DO kk = 1,ms
-
-                      ssnow%evapfbl(i,kk) = MIN( evapfb(i) * veg%froot(i,kk),      &
-                           MAX( 0.0, REAL( ssnow%wb(i,kk) ) -     &
-                           1.1 * soil%swilt(i) ) *                &
-                           soil%zse(kk) * density_liq )
+                local_fevc(i) = ( 1.0 - canopy%fwet(i)) * REAL( ecx(i) )
+                IF (local_fevc(i) > 0.0_r_2) THEN
+                
+                   ssnow%evapfbl(i,:) = transp_soil_water(dels, soil%swilt_vec(i,:),     &
+                            veg%froot(i,:), soil%zse_vec(i,:), local_fevc(i), ssnow%wbliq(i,:))
 
-                   ENDDO
                    IF (cable_user%soil_struc=='default') THEN
-                      canopy%fevc(i) = SUM(ssnow%evapfbl(i,:))*air%rlam(i)/dels
-                      ecx(i) = canopy%fevc(i) / (1.0-canopy%fwet(i))
-                   ELSEIF (cable_user%soil_struc=='sli') THEN
-                      canopy%fevc(i) = ecx(i)*(1.0-canopy%fwet(i))
-                   ENDIF
-
-                ENDIF
-
+                       canopy%fevc(i) = SUM(ssnow%evapfbl(i,:))*air%rlam(i)/dels
```

Comment:
not necessarily directly related to this PR - but the following section also links into internal inconsistencies.

I think we have:
1. `canopy%fevc` being rewritten possibly using values inconsistent with `ssnow%evapfbl` at line 611
2. `ssnow%evapfbl` being rewritten to an earlier value in the iteration cycle at line 645

Reply:
I know. I decided not to go there to try and keep this PR contained and not try to untangle the whole CABLE knot in one go :D. In other words, I was worried about falling into a big rabbit hole.